src/compiler/Stage/Emit/JavaScript.elm

```diff
@@ -45,6 +45,9 @@ emitExpr ( expr, _ ) =
         Plus e1 e2 ->
             "(" ++ emitExpr e1 ++ " + " ++ emitExpr e2 ++ ")"
 
+        ListConcat e1 e2 ->
+            "(" ++ emitExpr e1 ++ " ++ " ++ emitExpr e2 ++ ")"
```

Comment:
We'll have to do something different here. This is not valid JS. Also we'll have to break out of the "Lists represented by JS arrays" mindset, as hinted at in #29.

I think this could potentially be just a `Call` to `List.concat` of these two `Expr`s. I'm not sure whether to add this as an optimization step (`ListConcat e1 e2 -> Call (Call concat e1) e2`) or just here in Emit (`ListConcat e1 e2 -> emitExpr (Call ...)`)

Let me sleep on it :sweat_smile:

Reply:
oh ok, my bad, I misunderstood this part of the code :)

Reply:
no worries :)

Reply:
OK so I think the best way to go about this will be desugaring it into `Call`. That means `AST.Canonical.Expr` and `AST.Typed.Expr` won't have `Binop` constructor!